pkg/dhclient/dhclient.go

```diff
@@ -109,14 +73,51 @@ func HandlePacket(iface netlink.Link, packet Packet) error {
 		}
 	}
 
-	if ips := packet.DNS(); ips != nil {
-		rc := &bytes.Buffer{}
-		for _, ip := range ips {
-			rc.WriteString(fmt.Sprintf("nameserver %s\n", ip))
+	if ips := p.DNS(); ips != nil {
+		if err := WriteDNSSettings(ips); err != nil {
+			return err
 		}
-		if err := ioutil.WriteFile("/etc/resolv.conf", rc.Bytes(), 0644); err != nil {
+	}
+	return nil
+}
+
+// Configure6 adds IPv6 addresses, routes, and DNS servers to the system.
+func Configure6(iface netlink.Link, packet *dhcp6.Packet, iana *dhcp6opts.IANA) error {
+	p := NewPacket6(packet, iana)
+
+	l := p.Lease()
+	if l == nil {
+		return fmt.Errorf("no lease returned")
+	}
+
+	// Add the address to the iface.
+	dst := &netlink.Addr{
+		IPNet: &net.IPNet{
+			IP:   l.IP,
+			Mask: net.IPMask(net.ParseIP("ffff:ffff:ffff:ffff::")),
+		},
+		PreferedLft: int(l.PreferredLifetime.Seconds()),
+		ValidLft:    int(l.ValidLifetime.Seconds()),
+	}
+	if err := netlink.AddrReplace(iface, dst); err != nil {
+		if os.IsExist(err) {
+			return fmt.Errorf("add/replace %s to %v: %v", dst, iface, err)
+		}
+	}
+
+	if ips := p.DNS(); ips != nil {
+		if err := WriteDNSSettings(ips); err != nil {
 			return err
 		}
 	}
 	return nil
 }
+
+// WriteDNSSettings writes the given IPs as nameservers to resolv.conf.
+func WriteDNSSettings(ips []net.IP) error {
+	rc := &bytes.Buffer{}
+	for _, ip := range ips {
+		rc.WriteString(fmt.Sprintf("nameserver %s\n", ip))
```

Comment:
`IP.String()` can return output not suitable for resolv.conf, e.g. when the IP received from the DHCP server is malformed, or not an IPv4. This is OK for this PR, but for the future we may want to also enforce that `ip.To4() != nil` and error out if not.

I have created a separate issue to track all of these, https://github.com/u-root/u-root/issues/636

Reply:
To4() != nil is also not right, because v6. I think WriteDNSSettings should assume that the net.IP is correctly formatted. Whatever passes []net.IP here should be validating the IPs.

Reply:
yeah, I used `To4()` as an example but of course v6 and possibly other cases. But it's OK to assume that a correct `net.IP` is passed here. And we can always change our mind later